compiler/AST/expr.cpp

```diff
@@ -5479,7 +5487,7 @@ GenRet CallExpr::codegenPrimitive() {
       Type* dst = get(1)->typeInfo();
       Type* src = get(2)->typeInfo();
 
-      if (dst == src) {
+      if (dst == src && !(is_int_type(dst) || is_uint_type(dst)) ) {
```

Comment:
Hi - OK, in that case I don't think you're obligated to solve this issue, but at least put a comment explaining the problem.